src/main/java/ReviewScreens/RestaurantView.java

```diff
@@ -238,19 +261,61 @@ public RestaurantView(User user, Restaurant restaurant){
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        boolean check = true;
-        for(String reviewID : this.user.getPast_reviews()){
-            if (this.restaurant.getReviewIDs().contains(reviewID)) {
-                check = false;
-                break;
+        //If a user clicked the write review button
+        if(e.getActionCommand().equals("Write a Review")) {
+            //Check if they already reviewed this restaurant
+            boolean check = true;
+            for (String reviewID : this.user.getPast_reviews()) {
+                if (this.restaurant.getReviewIDs().contains(reviewID)) {
+                    check = false;
+                    break;
+                }
+            }
+            //Launch the proper view
+            if (check) {
+                WriteReviewScreen writeScreen = new WriteReviewScreen(this, this.writeReviewController,
+                        this.reviewGateway, this.userGateway, this.restaurantGateway, this.user, this.restaurant);
+            } else {
+                JOptionPane.showMessageDialog(this, "You've already written a review for this " +
+                        "restaurant. Please edit your previous review instead.");
+            }
+        //If an owner clicked edit
+        }else if(e.getActionCommand().equals("Edit your Restaurant")){
+            //Try to launch the appropriate view
+            try {
+                RestaurantDSGateway restaurantGateway = new FileRestaurant(RESTAURANT_DATABASE_NAME);
+                RestaurantPresenter restaurantPresenter = new RestaurantResponseFormatter(this);
+                RestaurantInputBoundary editRestaurant = new editRestaurant(restaurantGateway, restaurantPresenter);
+                RestaurantController restaurantController = new RestaurantController(editRestaurant);
+                EditRestaurantView editRestaurantView = new EditRestaurantView(restaurantController,
+                        (OwnerUser) this.user, this.restaurant);
+            }catch(IOException ex){
+                JOptionPane.showMessageDialog(this, "An error occurred. Please try again later.");
+                System.out.println("An error occurred. Please try again later.");
+                ex.printStackTrace();
+            }
+        //If an owner clicked delete
+        } else if (e.getActionCommand().equals("Delete Your Restaurant")) {
+            //Try to launch the appropriate view
+            try {
+                RestaurantDSGateway restaurantGateway = new FileRestaurant(RESTAURANT_DATABASE_NAME);
+                RestaurantDeletePresenter restaurantPresenter = new RestaurantDeleteResponseFormatter();
+                RestaurantDeleteInputBoundary deleteRestaurant = new deleteRestaurant(restaurantGateway,
+                        restaurantPresenter);
+                RestaurantDeleteController restaurantDeleteController =
+                        new RestaurantDeleteController(deleteRestaurant);
```

Comment:
Sorry for the change but my current active pull request removes the need for creating the controllers outside the view, and with the start of GUI connectivity, we also need to pass through previous frames.